File: engine/model/src/main/scala/tools/packet/pet/PetExceptionList.scala
package tools.packet.pet

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class PetExceptionList(private var _characterId: Int, private var _petId: Int, private var _petIndex: Byte,
                       private var _exclusionList: java.util.List[java.lang.Integer]) extends PacketInput {
  def characterId: Int = _characterId

  def petId: Int = _petId

  def petIndex: Byte = _petIndex

  def exclusionList: java.util.List[java.lang.Integer] = _exclusionList

  override def opcode(): SendOpcode = SendOpcode.PET_EXCEPTION_LIST
}

File: engine/engine-model/src/main/java/tools/packet/pet/PetExceptionList.java
package tools.packet.pet;

import java.util.List;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record PetExceptionList(Integer characterId, Integer petId, Byte petIndex,
                               List<Integer> exclusionList) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.PET_EXCEPTION_LIST;
   }
}